cloudtools/aws/spot.py

```diff
@@ -240,29 +242,42 @@ def get_current_spot_prices(connection, product_description, start_time=None,
         yesterday = now - timedelta(hours=24)
         start_time = yesterday.isoformat() + "Z"
 
-    while True:
-        log.debug("getting spot prices for instance_type %s from %s "
-                  "(next_token %s)", instance_type, start_time, next_token)
+    if ignored_availability_zones is None:
+        ignored_availability_zones = []
+    all_zones = set([az.name for az in connection.get_all_zones()])
+    useful_zones = all_zones - set(ignored_availability_zones)
+    remaining = useful_zones
+    log.debug("getting spot prices for instance_type %s in %s, from %s",
+              instance_type, sorted(remaining), start_time)
+    while remaining:
         all_prices = connection.get_spot_price_history(
             product_description=product_description,
             instance_type=instance_type,
             start_time=start_time,
+            max_results=50,
             next_token=next_token,
         )
```

Comment:
does get_spot_price_history return data in ascending or descending chronological order?